graphdatascience/tests/integration/test_nx_loader.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+from graphdatascience.graph_data_science import GraphDataScience
+from graphdatascience.server_version.server_version import ServerVersion
+
+GRAPH_NAME = "banana"
+
+pytest.importorskip("networkx")
+
+import networkx as nx  # noqa: E402
+
+
+@pytest.fixture
+def undirected_nx_G() -> nx.Graph:
+    nx_G = nx.Graph()
+
+    nx_G.add_node(1, labels=["N"], foo=1)
+    nx_G.add_node(42, labels=["N", "M"], foo=2)
+    nx_G.add_node(1337, labels=["N"], foo=3)
+    nx_G.add_edge(1, 42, relationshipType="R", weight=0.4)
+    nx_G.add_edge(1, 1337, relationshipType="R", weight=1.4)
+    nx_G.add_edge(42, 1337, relationshipType="R", weight=1.1)
+
+    return nx_G
+
+
+@pytest.fixture
+def directed_nx_G() -> nx.Graph:
+    nx_G = nx.DiGraph()
+
+    nx_G.add_node(1, labels=["N"], foo=1)
+    nx_G.add_node(42, labels=["N", "M"], foo=2)
+    nx_G.add_node(1337, labels=["N"], foo=3)
+    nx_G.add_edge(1, 42, relationshipType="R", weight=0.4)
+    nx_G.add_edge(1, 1337, relationshipType="R", weight=1.4)
+    nx_G.add_edge(42, 1337, relationshipType="R", weight=1.1)
+
+    return nx_G
+
+
+@pytest.mark.compatible_with(min_inclusive=ServerVersion(2, 1, 0))
+def test_undirected_nx(gds: GraphDataScience, undirected_nx_G: nx.Graph) -> None:
+    G = gds.graph.networkx.load(undirected_nx_G, GRAPH_NAME)
+
+    assert G.name() == GRAPH_NAME
+
+    assert G.node_count() == 3
+    assert set(G.node_labels()) == {"N", "M"}
+    assert G.node_properties("N") == ["foo"]
+    assert G.node_properties("M") == ["foo"]
+
+    assert G.relationship_count() == 6
+    assert G.relationship_types() == ["R"]
+    assert G.relationship_properties("R") == ["weight"]
+
+    G.drop()
+
+
+@pytest.mark.compatible_with(min_inclusive=ServerVersion(2, 1, 0))
+def test_undirected_nx_without_Arrow(gds_without_arrow: GraphDataScience, undirected_nx_G: nx.Graph) -> None:
```

Comment:
arguably we only need to test the non arrow option here as the arrow vs non-arrow construct is tested at another place

Reply:
True, but it's so cheap :)